### PR TITLE
Adding SQL Settings specs

### PR DESCRIPTION
**Pull request by Tokesh**
### Description
adding sql settings specs

### Issues Resolved
[#235]

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -36,7 +36,6 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Added JsonSchemaValidator, a wrapper for AJV ([#364](https://github.com/opensearch-project/opensearch-api-specification/issues/364))
 - Added support for `application/cbor` responses ([#371](https://github.com/opensearch-project/opensearch-api-specification/pull/371))
 - Added `/_plugins/_sql`, `close`, `explain` and `stats` ([#379](https://github.com/opensearch-project/opensearch-api-specification/pull/379))
-- Added tests for SQL namespace ([#379](https://github.com/opensearch-project/opensearch-api-specification/pull/379))
```
Comment (Tokesh) [inline CHANGELOG.md:39]: as you mentioned at previous PR @dblock 
i deleted this bullet point at changelog

File: tests/sql/query.yaml [hunk 2]
```diff
@@ -17,6 +17,8 @@ chapters:
   - synopsis: Get SQL query
     path: /_plugins/_sql
     method: POST
+    parameters:
+      sanitize: false
```
Comment (Tokesh) [inline tests/sql/query.yaml:21]: as you mentioned at previous PR @dblock
i added parameters with sanitize for one test

**Comment by Tokesh** [comment]
But i have a question. How i can describe this request body in specs? When user can input 2 different formats of request body
image

File: spec/_superseded_operations.yaml [hunk 3]
```diff
@@ -586,7 +586,7 @@ $schema: ./json_schemas/_superseded_operations.schema.yaml
   operations:
     - POST
 /_opendistro/_sql/settings:
-  superseded_by: /_plugins/_sql/settings
+  superseded_by: /_plugins/_query/settings
```
Comment (nhtruong) [inline spec/_superseded_operations.yaml:589]: Does this mean `/_plugins/_sql/settings` actually doesn't exist?

Comment (Tokesh) [reply spec/_superseded_operations.yaml:589]: Yeap! 
image
image

File: tests/sql/settings.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+            sql:
+              enabled: 'true'
+              slowlog: '2'
+              cursor:
+                keep_alive: 1m
+            ppl:
+              enabled: 'true'
+            query:
+              memory_limit: '80%'
+              size_limit: '200'
```
Comment (dblock) [inline tests/sql/settings.yaml:22]: Are these all strings (`true`, `200`)? Logically shouldn't they be booleans/numbers?

**Review by dblock — APPROVED** [review]

File: spec/schemas/sql._common.yaml [hunk 5]
```diff
@@ -85,4 +85,59 @@ components:
...
+      properties:
+        enabled:
+          type: string
+    Query:
+      type: object
+      properties:
+        memory_limit:
+          type: string
+        size_limit:
+          type: string
```
Comment (dblock) [inline spec/schemas/sql._common.yaml:128]: Double-checking that those are actually strings?

File: spec/schemas/sql._common.yaml [hunk 6]
```diff
@@ -85,4 +85,59 @@ components:
...
+
+    Sql:
+      type: object
+      properties:
+        enabled:
+          type: string
```
Comment (dblock) [inline spec/schemas/sql._common.yaml:134]: Boolean?

**Review by dblock — CHANGES_REQUESTED** [review]
Let's make sure the types are correct.

**Comment by Tokesh** [comment]
> Let's make sure the types are correct.

I rechecked it
image

File: spec/namespaces/sql.yaml [hunk 7]
```diff
@@ -83,6 +98,13 @@ paths:
           $ref: '#/components/responses/sql.post_stats@200'
 components:
   parameters:
+    sql.settings::query.format:
+      name: format
+      in: query
+      description: A short version of the Accept header, e.g. json, yaml.
+      schema:
+        type: string
+        description: A short version of the Accept header, e.g. json, yaml.
```
Comment (dblock) [inline spec/namespaces/sql.yaml:107]: This should only exist above I believe, not in `schema`.

File: spec/namespaces/sql.yaml [hunk 8]
```diff
@@ -184,6 +214,12 @@ components:
             $ref: '../schemas/sql._common.yaml#/components/schemas/SqlStats'
       required: true
   responses:
+    sql.settings@200:
+      description: ''
```
Comment (dblock) [inline spec/namespaces/sql.yaml:218]: Add a description or remove the field.

**Review by dblock — CHANGES_REQUESTED** [review]
Thanks! It's close. See comments, look through all unresolved conversations. The types look suspicious.

File: spec/schemas/sql._common.yaml (hunk 5 above)
Comment (Tokesh) [reply spec/schemas/sql._common.yaml:133]: This API accept string and also numbers. 
image
image

**Comment by Tokesh** [comment]
> Thanks! It's close. See comments, look through all unresolved conversations. The types look suspicious.

Okkie! I resolved all comments
Also, I attached a few screenshots related to types.

File: spec/namespaces/sql.yaml [hunk 9]
```diff
@@ -4,6 +4,21 @@ info:
   description: OpenSearch SQL API
   version: 1.0.0
 paths:
+  /_plugins/_query/settings:
```
Comment (dblock) [inline spec/namespaces/sql.yaml:7]: This is new, let's add "Added `/_plugins/_query/settings`" to the CHANGELOG?

**Review by dblock — CHANGES_REQUESTED** [review]
Looks good. One ask for CHANGELOG and we can merge.

File: tests/sql/settings.yaml [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+chapters:
+  - synopsis: Update SQL settings.
+    path: /_plugins/_query/settings
+    method: PUT
+    request_body:
+      payload:
+        transient: 
+          plugins:
+            sql:
+              enabled: 'true'
```
Comment (dblock) [inline tests/sql/settings.yaml:14]: This should be not quoted since it's a boolean?

**Review by dblock — CHANGES_REQUESTED** [review]
Question on boolean:

File: tests/sql/settings.yaml (hunk 10 above)
Comment (Tokesh) [reply tests/sql/settings.yaml]: Yeap! Sure, i fixed it

**Comment by Tokesh** [comment]
@dblock @nhtruong ready for your review!

**Review by dblock — APPROVED** [review]

**Pull request merged**